ext/kv/01_db.ts

```diff
@@ -419,11 +421,26 @@ class AtomicOperation {
   }
 
   async commit(): Promise<Deno.KvCommitResult | Deno.KvCommitError> {
+    let mutations = this.#mutations;
+
+    // Translate expireIn to expireAt, without overwriting `this.#mutations`
+    if (mutations.findIndex((x) => x[3] !== undefined) !== -1) {
+      const now = Date.now();
+      mutations = mutations.map(([key, type, value, expireIn]) => {
+        return [
+          key,
+          type,
+          value,
+          expireIn !== undefined ? now + expireIn : undefined,
+        ];
+      });
+    }
```

Comment:
We do two loops over the array here (findIndex and map) - can you reduce this to one?

Reply:
If we do one loop (construct the patched `mutations` array directly), there will be unnecessary allocation if there is no keys with expiration in this atomic op. Unless we overwrite `this.#mutations` which I don't want to do (would be confusing if the user accidentally committed twice).

Reply:
I pushed a patch to simplify the logic